--- /dev/null
+++ b/build_files/build_environment/patches/ch_legacy_python.diff
@@ -0,0 +1,11 @@
+--- a/PCbuild/openssl.vcxproj
++++ b/PCbuild/openssl.vcxproj
+@@ -67,7 +67,7 @@
+ set VCINSTALLDIR=$(VCInstallDir)
+ if not exist "$(IntDir.TrimEnd('\'))" mkdir "$(IntDir.TrimEnd('\'))"
+ cd /D "$(IntDir.TrimEnd('\'))"
+-$(Perl) "$(opensslDir)\configure" $(OpenSSLPlatform) no-asm
++$(Perl) "$(opensslDir)\configure" $(OpenSSLPlatform) no-asm no-tests
+ nmake
+ </NMakeBuildCommandLine>
+   </PropertyGroup>
--- a/build_files/build_environment/cmake/python.cmake
+++ b/build_files/build_environment/cmake/python.cmake
@@ -36,6 +36,11 @@
   cmake_to_dos_path(${ZLIB_SOURCE_FOLDER} ZLIB_SOURCE_FOLDER_DOS)
   cmake_to_dos_path(${SSL_SOURCE_FOLDER} SSL_SOURCE_FOLDER_DOS)
   cmake_to_dos_path(${DOWNLOADS_EXTERNALS_FOLDER} DOWNLOADS_EXTERNALS_FOLDER_DOS)
+
+  set(CH_LEGACY_PYTHON_PATCH)
+  if(CH_BLENDER_LEGACY_CPU)
+    set(CH_LEGACY_PYTHON_PATCH && ${PATCH_CMD} -p1 -d ${BUILD_DIR}/python/src/external_python < ${PATCH_DIR}/ch_legacy_python.diff)
+  endif()
 
   ExternalProject_Add(external_python
     URL file://${PACKAGE_DIR}/${PYTHON_FILE}
@@ -56,6 +61,7 @@
       ${PATCH_CMD} --verbose -p1 -d
         ${BUILD_DIR}/python/src/external_python <
         ${PATCH_DIR}/${PYTHON_PATCH_FILE}
+      ${CH_LEGACY_PYTHON_PATCH}
 
     CONFIGURE_COMMAND echo "."
 
