--- a/build_files/cmake/platform/platform_win32.cmake
+++ b/build_files/cmake/platform/platform_win32.cmake
@@ -670,7 +670,11 @@
     set(BOOST_DEBUG_POSTFIX "vc143-mt-gyd-a64-${BOOST_VERSION}")
     set(BOOST_PREFIX "")
   else()
-    set(BOOST_POSTFIX "vc142-mt-x64-${BOOST_VERSION}")
+    if(CH_BLENDER_LEGACY_CPU)
+      set(BOOST_POSTFIX "vc143-mt-x64-${BOOST_VERSION}")
+    else()
+      set(BOOST_POSTFIX "vc142-mt-x64-${BOOST_VERSION}")
+    endif()
     set(BOOST_DEBUG_POSTFIX "vc142-mt-gyd-x64-${BOOST_VERSION}")
     set(BOOST_PREFIX "")
   endif()
@@ -694,7 +698,7 @@
     # This is file new in 3.4 if it does not exist, assume we are building against 3.3 libs
     # Note, as ARM64 was introduced in 4.x, this check is not needed
     set(BOOST_34_TRIGGER_FILE ${BOOST_LIBPATH}/${BOOST_PREFIX}boost_python${_PYTHON_VERSION_NO_DOTS}-${BOOST_DEBUG_POSTFIX}.lib)
-    if(NOT EXISTS ${BOOST_34_TRIGGER_FILE})
+    if(NOT EXISTS ${BOOST_34_TRIGGER_FILE} AND NOT CH_BLENDER_LEGACY_CPU)
       set(BOOST_DEBUG_POSTFIX "vc142-mt-gd-x64-${BOOST_VERSION}")
       set(BOOST_PREFIX "lib")
     endif()
